### PR TITLE
add droplet

### PR DESCRIPTION
**Pull request by cyLi-Tiger**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context

Add the droplet chip support in DIOPI main branch.

## Description

- Add CMakeLists.txt for droplet for building.
- Implement the export functions and export runtimes for droplet chip for DIOPI support.

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/droplet/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.5)
+project(droplet_impl)
+
+option(TEST "whether to test by using conformance test" ON)
+option(DEBUG "whether to use debug" OFF)
```
Comment (yangbofun) [inline impl/droplet/CMakeLists.txt:5]: 这个DEBUG删掉吧，还有下面的DEBUG的判断。 上面的主CMakeLists.txt已经有了。

File: impl/droplet/convert_config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+- common_config:
+    dtype: (int64)->int32, (float64)->float32
+    layout: NCHW
+
+- diopiAdd:
```
Comment (yangbofun) [inline impl/droplet/convert_config.yaml:5]: common_config里面写了，这里add和common_config内容一样，就不用写了。

File: impl/scripts/build_impl.sh [hunk 3]
```diff
@@ -34,6 +34,8 @@ case $1 in
     (cd third_party/mmcv_diopi && rm -rf build && mkdir build \
         && MMCV_WITH_DIOPI=1 MMCV_WITH_OPS=1 python setup.py build_ext -i) \
     || exit -1;;
+  droplet)
+    mkdir build && cd build && cmake .. -DIMPL_OPT=droplet  -DTEST=${DIOPI_BUILD_TESTRT} && make -j8 || exit -1;;
```
Comment (yangbofun) [inline impl/scripts/build_impl.sh:38]: 看下main分支，这里-DTEST直接写死为ON

File: impl/droplet/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        --diopi_dir=${CMAKE_SOURCE_DIR}/../ --output_dir=${CMAKE_SOURCE_DIR}/../adaptor
+        --config_device=droplet)
+    set(USE_ADAPTOR "true")
+    add_dependencies(diopirt adaptor_code_gen)
+    add_definitions(-DTEST_USE_ADAPTOR)
+else()
+    set(USE_ADAPTOR "false")
+endif()
+
+target_link_libraries(${DIOPIRT} PRIVATE diopirt)
```
Comment (yewentao256) [inline impl/droplet/test/CMakeLists.txt:50]: 看这里能不能改个变量名，两个rt容易引起混淆

Comment (yewentao256) [reply impl/droplet/test/CMakeLists.txt:50]: 此外可能需要在7.5 gcc上测试一下，能否正常找到并链接弱符号（如果不行考虑加-Wl,--no-as-needed）

File: impl/droplet/test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    COMMAND ln -f ${FUNCTION_SAVE_PATH}/export_functions.cpp ${PROJECT_SOURCE_DIR}/test
+    DEPENDS test_code_gen)
+
+# file(MAKE_DIRECTORY ${FUNCTION_SAVE_PATH})
+file(TOUCH export_functions.cpp)
+set(FUNCTIONS_SRC
+    export_functions.cpp
+)
+
+pybind11_add_module(${DIOPIFUNCTIONS} SHARED ${FUNCTIONS_SRC})
```
Comment (yewentao256) [inline impl/droplet/test/CMakeLists.txt:69]: 同上

File: impl/droplet/test/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        --diopi_dir=${CMAKE_SOURCE_DIR}/../ --output_dir=${CMAKE_SOURCE_DIR}/../adaptor
+        --config_device=droplet)
+    set(USE_ADAPTOR "true")
+    add_dependencies(diopiruntime adaptor_code_gen)
+    add_definitions(-DTEST_USE_ADAPTOR)
+else()
+    set(USE_ADAPTOR "false")
+endif()
+
+target_link_libraries(${DIOPIEXPORTRT} PRIVATE diopiruntime)
```
Comment (yewentao256) [inline impl/droplet/test/CMakeLists.txt:50]: 纯大写变量名，单词间建议加上下划线区分

File: impl/droplet/test/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    DEPENDS test_code_gen)
+
+# file(MAKE_DIRECTORY ${FUNCTION_SAVE_PATH})
+file(TOUCH export_functions.cpp)
+set(FUNCTIONS_SRC
+    export_functions.cpp
+)
+
+pybind11_add_module(${DIOPI_EXPORT_FUNCTIONS} SHARED ${FUNCTIONS_SRC})
+target_link_libraries(${DIOPI_EXPORT_FUNCTIONS} PRIVATE -Wl,--no-as-needed diopiruntime -Wl,--no-as-needed)
```
Comment (yewentao256) [inline impl/droplet/test/CMakeLists.txt:70]: target_link_libraries(${DIOPI_EXPORT_FUNCTIONS} PRIVATE -Wl,--no-as-needed diopiruntime -Wl,--as-needed)

**Review by yewentao256 — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**